includes/ting_search_autocomplete.admin.inc

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * @file
+ * Admin settings for bibliotek.dk autocomplete.
+ */
+
+function ting_search_autocomplete_admin() {
+
+  $base_settings['index'] = NULL;
+  $base_settings['facetIndex'] = NULL;
+  $base_settings['filterQuery'] = NULL;
+  $base_settings['sort'] = 'count';
+
+  $base_settings['agency']  = variable_get('ting_agency', NULL);
+  $base_settings['profile'] = variable_get('ting_search_profile', NULL);
+  $base_settings['maxSuggestions'] = 10;
+  $base_settings['maxTime'] = 2000;
+
+  $base_settings['highlight'] = NULL;
+  $base_settings['highlightPre'] = NULL;
+  $base_settings['highlightPost'] = NULL;
+  $base_settings['minimumString'] = 3;
+
+  $defaults = variable_get('ting_search_autocomplete_settings', array());
+
+  foreach ($base_settings as $key => $base_setting) {
+    $default_settings[$key] = (!empty($defaults[$key])) ? $defaults[$key] : $base_setting;
+  }
```

Comment:
All these defaults are defined here as well as other places and merged together with the settings.

This could be handled more elegantly and consistently by defining one function that merges defaults with settings and returns the resulting array.